Mathlib/Data/Complex/Basic.lean

```diff
@@ -970,353 +959,7 @@ theorem im_eq_sub_conj (z : ℂ) : (z.im : ℂ) = (z - conj z) / (2 * I) := by
-  ⟨fun f => ⟨limAux f, equiv_limAux f⟩⟩
-
-open CauSeq
-
-theorem lim_eq_lim_im_add_lim_re (f : CauSeq ℂ Complex.abs) :
-    lim f = ↑(lim (cauSeqRe f)) + ↑(lim (cauSeqIm f)) * I :=
-  lim_eq_of_equiv_const <|
-    calc
-      f ≈ _ := equiv_limAux f
-      _ = CauSeq.const Complex.abs (↑(lim (cauSeqRe f)) + ↑(lim (cauSeqIm f)) * I) :=
-        CauSeq.ext fun _ =>
-          Complex.ext (by simp [limAux, cauSeqRe, ofReal']) (by simp [limAux, cauSeqIm, ofReal'])
-#align complex.lim_eq_lim_im_add_lim_re Complex.lim_eq_lim_im_add_lim_re
-
-theorem lim_re (f : CauSeq ℂ Complex.abs) : lim (cauSeqRe f) = (lim f).re := by
-  rw [lim_eq_lim_im_add_lim_re]; simp [ofReal']
-#align complex.lim_re Complex.lim_re
-
-theorem lim_im (f : CauSeq ℂ Complex.abs) : lim (cauSeqIm f) = (lim f).im := by
-  rw [lim_eq_lim_im_add_lim_re]; simp [ofReal']
-#align complex.lim_im Complex.lim_im
-
-theorem isCauSeq_conj (f : CauSeq ℂ Complex.abs) :
-    IsCauSeq Complex.abs fun n => conj (f n) := fun ε ε0 =>
-  let ⟨i, hi⟩ := f.2 ε ε0
-  ⟨i, fun j hj => by
-    rw [← RingHom.map_sub, abs_conj]; exact hi j hj⟩
-#align complex.is_cau_seq_conj Complex.isCauSeq_conj
-
-/-- The complex conjugate of a complex Cauchy sequence, as a complex Cauchy sequence. -/
-noncomputable def cauSeqConj (f : CauSeq ℂ Complex.abs) : CauSeq ℂ Complex.abs :=
-  ⟨_, isCauSeq_conj f⟩
-#align complex.cau_seq_conj Complex.cauSeqConj
-
-theorem lim_conj (f : CauSeq ℂ Complex.abs) : lim (cauSeqConj f) = conj (lim f) :=
-  Complex.ext (by simp [cauSeqConj, (lim_re _).symm, cauSeqRe])
-    (by simp [cauSeqConj, (lim_im _).symm, cauSeqIm, (lim_neg _).symm]; rfl)
-#align complex.lim_conj Complex.lim_conj
-
-/-- The absolute value of a complex Cauchy sequence, as a real Cauchy sequence. -/
-noncomputable def cauSeqAbs (f : CauSeq ℂ Complex.abs) : CauSeq ℝ abs' :=
-  ⟨_, isCauSeq_abs f.2⟩
-#align complex.cau_seq_abs Complex.cauSeqAbs
-
-theorem lim_abs (f : CauSeq ℂ Complex.abs) : lim (cauSeqAbs f) = Complex.abs (lim f) :=
-  lim_eq_of_equiv_const fun ε ε0 =>
-    let ⟨i, hi⟩ := equiv_lim f ε ε0
-    ⟨i, fun j hj => lt_of_le_of_lt (Complex.abs.abs_abv_sub_le_abv_sub _ _) (hi j hj)⟩
-#align complex.lim_abs Complex.lim_abs
-
-variable {α : Type*} (s : Finset α)
-
-@[simp, norm_cast]
-theorem ofReal_prod (f : α → ℝ) : ((∏ i in s, f i : ℝ) : ℂ) = ∏ i in s, (f i : ℂ) :=
-  map_prod ofReal _ _
-#align complex.of_real_prod Complex.ofReal_prod
-
-@[simp, norm_cast]
-theorem ofReal_sum (f : α → ℝ) : ((∑ i in s, f i : ℝ) : ℂ) = ∑ i in s, (f i : ℂ) :=
-  map_sum ofReal _ _
-#align complex.of_real_sum Complex.ofReal_sum
-
-@[simp]
-theorem re_sum (f : α → ℂ) : (∑ i in s, f i).re = ∑ i in s, (f i).re :=
-  reAddGroupHom.map_sum f s
-#align complex.re_sum Complex.re_sum
-
-@[simp]
-theorem im_sum (f : α → ℂ) : (∑ i in s, f i).im = ∑ i in s, (f i).im :=
-  imAddGroupHom.map_sum f s
-#align complex.im_sum Complex.im_sum
-
 end Complex
+
+assert_not_exists Multiset
+assert_not_exists Algebra
```

Comment:
In the long run I'd be tempted to move the `Module` and `Algebra` structure into `Basic`; it's pretty annoying to find that structure missing (almost as bad as needing a separate import to make polynomials commutative...).
I think for that to be acceptable, we probably need to slim down `Algebra/Algebra/Basic.lean`

Either way, `assert_not_exists`s are harmless, and just force the next reviewer to actually think about what's going on :)